packages/swingset-liveslots/test/virtual-objects/test-empty-data.js

```diff
@@ -0,0 +1,34 @@
+import test from 'ava';
+import '@endo/init/debug.js';
+
+import { makeFakeVirtualObjectManager } from '../../tools/fakeVirtualSupport.js';
+
+test('non-object initial data message', async t => {
+  const vom = makeFakeVirtualObjectManager();
+  // 'goodInit' is () => ({ value: 0 })
+  const goodInit = () => ({ value: 0 });
+
+  // 'badInit' is  () =>  { value: 0 }
+  //
+  // badInit is subtly wrong: it returns undefined, and includes an
+  // unused "value:" label, plus a side-effect-free evaluation of a 0
+  // constant. A linter might catch this, but maybe you didn't run
+  // one.
+  //
+  // I frequently mean to write goodInit and write badInit
+  // instead. It's such an easy mistake to make, so the VOM's instance
+  // maker will catch it and complain.
+  //
+  // note: it is not easy to convince our editors and linters and
+  // prettiers to leave this erroneous code in place
+
+  // eslint-disable-next-line
+  /* prettier-ignore */ const badInit = () => { value: 0 };
+
+  const behavior = {};
+  const makeGoodThing = vom.defineKind('goodthing', goodInit, behavior);
+  const makeBadThing = vom.defineKind('badthing', badInit, behavior);
+  makeGoodThing();
+  const m = s => ({ message: s });
+  t.throws(() => makeBadThing(), m(/initial data must be object, not /));
```

Comment:
```suggestion
  t.throws(() => makeBadThing(), m(/^initial data must be object, not /));
```
Should we just introduce helpers for this?
```js
const throwsErrorPrefix = (t, fn, prefix) =>
  t.throws(fn, { message: err => err.message.startsWith(prefix) });
const throwsAsyncPrefix = (t, fn, prefix) =>
  t.throwsAsync(fn, { message: err => err.message.startsWith(prefix) });
```
```suggestion
  throwsErrorPrefix(() => makeBadThing(), 'initial data must be object, not ');
```

Reply:
maybe, I like simple assertion statements, and I like short lines, and adding a helper with a name specific enough to be readable (without consulting the definition) tends to require a long-ish name

Also I'm not sure that anchoring the regexp is super necessary, the fact that it throws is the main thing. If the error message is changed slightly, I don't need to update the test. (I'm being more casual about asserting the contents of error messages than I used to, partly because I'm in a hurry, partly because those messges frequently include the specific object details at the far end, which are messy and sometimes hard to predict ahead of time from a test, like the ones that include a vref).

I suspect I'd be more inclined to make a `mp` helper than a wrapper around `t.throws`, so it's still obvious which lines of the test are performing assertions ("everything that starts with `t.`").

Reply:
> it's still obvious which lines of the test are performing assertions ("everything that starts with `t.`")

Interesting, but somewhat at odds with introducing non-invasive helpers that wrap multiple test assertions or combinations of system-under-test interactions and test assertions into a logical unit such as
* [`kpidRejected`](https://github.com/Agoric/agoric-sdk/blob/ed68edf80ceb09e619877e820cf57266948470a2/packages/SwingSet/test/metering/test-dynamic-vat-metered.js#L93)
* [`checkPromises`](https://github.com/Agoric/agoric-sdk/blob/ed68edf80ceb09e619877e820cf57266948470a2/packages/SwingSet/test/test-kernel.js#L16)
* [`expectError`](https://github.com/Agoric/agoric-sdk/blob/ed68edf80ceb09e619877e820cf57266948470a2/packages/swingset-liveslots/test/test-liveslots.js#L30)
* [`weakKeyCheck`](https://github.com/Agoric/agoric-sdk/blob/ed68edf80ceb09e619877e820cf57266948470a2/packages/swingset-liveslots/test/virtual-objects/test-cease-recognition.js#L26)
* [`amountsEqual`](https://github.com/Agoric/agoric-sdk/blob/ed68edf80ceb09e619877e820cf57266948470a2/packages/zoe/test/unitTests/contractSupport/test-percentSupport.js#L13) ([and](https://github.com/Agoric/agoric-sdk/blob/ed68edf80ceb09e619877e820cf57266948470a2/packages/zoe/test/unitTests/contractSupport/test-ratio.js#L31) [clones](https://github.com/Agoric/agoric-sdk/blob/ed68edf80ceb09e619877e820cf57266948470a2/packages/zoe/test/zoeTestHelpers.js#L7))
* etc.

Would you settle for a naming convention in support of generalizing "_everything that starts with `t.`_" to e.g. "_everything that starts with `t.` **or** sends `t` as the first argument to an `assert…`[^1] function_"?

[^1]: alternatively bikeshedding "`assert…`" to "`check…`" or "`validate…`" or whatever, keeping in mind the [endo `Checker` explanation](https://github.com/endojs/endo/blob/916f6d42270b282641d3299abdb4ca9c835a039b/packages/pass-style/src/types.js#L132-L144)

Reply:
Good argument (and I'm impressed by your citations), and I agree we could use a convention. I haven't thought through what a good one would be.

I guess helpers which only exist to let simple things be simple (by which I probably mean "don't insert gratuitous whitespace) don't feel like they're worth creating a new name. Even without looking up how `m` is defined, you can tell that `t.throws( ... m()` is doing a throwsAsync, and you could deduce that `m` has something to do with the expectations. `throwsErrorPrefix` could, in the right frame of confusion, sound like it throws an error named "prefix".

But obviously I've been pretty casual with the helper function creation machine in the past.

I think I like the use of an `assert` prefix for things that close over `t` and then perform an assertion. I can see `assertKpidRejected` and `assertPromisesMatched` as names for the first two. And then `assertThrowsErrorPrefix` might be a decent name for the `m()`-using helper (although a bit wordy).

I'm going to prioritize brevity for now, but I'll try to keep this kind of convention in mind for future tests.